src/include/mpir_hw_topo.h

```diff
@@ -0,0 +1,232 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
+/*
+ *  (C) 2019 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ *
+ */
+
+#ifndef MPIR_HW_TOPO_H_INCLUDED
+#define MPIR_HW_TOPO_H_INCLUDED
+
+/*
+ * Node hardware object types
+ */
+typedef enum {
+    MPIR_NODE_OBJ_TYPE__NONE = -1,
+    MPIR_NODE_OBJ_TYPE__MACHINE,
+    MPIR_NODE_OBJ_TYPE__PACKAGE,
+    MPIR_NODE_OBJ_TYPE__CORE,
+    MPIR_NODE_OBJ_TYPE__PU,
+    MPIR_NODE_OBJ_TYPE__L1CACHE,
+    MPIR_NODE_OBJ_TYPE__L2CACHE,
+    MPIR_NODE_OBJ_TYPE__L3CACHE,
+    MPIR_NODE_OBJ_TYPE__L4CACHE,
+    MPIR_NODE_OBJ_TYPE__L5CACHE,
+    MPIR_NODE_OBJ_TYPE__L1ICACHE,
+    MPIR_NODE_OBJ_TYPE__L2ICACHE,
+    MPIR_NODE_OBJ_TYPE__L3ICACHE,
+    MPIR_NODE_OBJ_TYPE__GROUP,
+    MPIR_NODE_OBJ_TYPE__NUMANODE,
+    MPIR_NODE_OBJ_TYPE__BRIDGE,
+    MPIR_NODE_OBJ_TYPE__PCI_DEVICE,
+    MPIR_NODE_OBJ_TYPE__OS_DEVICE,
+    MPIR_NODE_OBJ_TYPE__MISC,
+    MPIR_NODE_OBJ_TYPE__MAX
+} MPIR_Node_obj_type;
+
+/*
+ * Node hardware OS device object types
```

Comment:
I would suggest add note that these are sub-types for `MPIR_NODE_OBJ_TYPE__OS_DEVICE` -- unless I interpreted wrong.